nbtlib/path.py

```diff
@@ -0,0 +1,278 @@
+    except ValueError:
+        return False
+
+
+def parse_accessors(path):
+    try:
+        parser = Parser(tokenize(path))
+    except InvalidLiteral:
+        return ()
+
+    while True:
+        try:
+            tag = parser.parse()
+        except InvalidLiteral as exc:
+            raise InvalidPath(f'Invalid path at position {exc.args[0][0]}') from exc
+
+        if isinstance(tag, String):
+            if parser.current_token.type == 'QUOTED_STRING':
+                yield NamedKey(tag[:])
+            else:
+                yield from (NamedKey(key) for key in tag.split('.') if key)
+
+        elif isinstance(tag, List):
+            if not tag:
+                yield ListIndex(index=None)
+            elif len(tag) != 1:
+                raise InvalidPath('Brackets should only contain one element')
+            elif issubclass(tag.subtype, Compound):
+                yield ListIndex(index=None)
+                yield CompoundMatch(tag[0])
+            elif issubclass(tag.subtype, Int) or can_be_converted_to_int(tag[0]):
+                yield ListIndex(int(tag[0]))
+            else:
+                raise InvalidPath('Brackets should only contain an integer or a compound')
+
+        elif isinstance(tag, Compound):
+            yield CompoundMatch(tag)
+
+        elif parser.current_token.type == 'NUMBER':
+            yield from (NamedKey(key) for key in parser.current_token.value.split('.') if key)
+
+        else:
+            raise InvalidPath(f'Invalid path element {tag}')
+
+        try:
+            parser.next()
+        except InvalidLiteral:
+            break
+
+
+def extend_accessors(accessors, new_accessor):
+    if isinstance(new_accessor, CompoundMatch) and accessors:
+        *except_last, last_accessor = accessors
+
+        if isinstance(last_accessor, CompoundMatch):
+            return tuple(except_last) + (
+                CompoundMatch(new_accessor.compound.with_defaults(last_accessor.compound)),
+            )
+        if isinstance(last_accessor, ListIndex) and last_accessor.index is not None:
+            raise InvalidPath('Can\'t match a compound on list items '
+                              f'selected with {last_accessor!r}')
+    return accessors + (new_accessor,)
+
+
+class NamedKey(NamedTuple):
+    key: str
+
+    UNQUOTED_REGEX = re.compile(r'^[a-zA-Z0-9_]+$')
+
+    def get(self, tags):
+        return [(tag, tag[self.key]) for _, tag in tags if self.key in tag]
+
+    def set(self, tags, value):
+        for parent, _ in tags:
+            parent[self.key] = value
+
+    def delete(self, tags):
+        for parent, _ in tags:
+            if self.key in parent:
+                del parent[self.key]
+
+    def __str__(self):
+        return (
+            self.key if self.UNQUOTED_REGEX.match(self.key) else
+            '"' + self.key.replace('"', '\\"') + '"'
+        )
+
+
+class ListIndex(NamedTuple):
+    index: Optional[int]
+
+    def get(self, tags):
+        if self.index is None:
```

Comment:
please add a `tags=[i for i in tags if isinstance(i,(List,Array))]`
at the beginning of the function.

Reply:
Fixed in 53fb2bbb72d5b96635a889d09a4e97d904487500